public/app/features/query-library/api/mappers.test.ts

```diff
@@ -0,0 +1,18 @@
+import { parseCreatedByValue } from './mappers';
+
+describe('mappers', () => {
+  describe('parseCreatedByValue', () => {
+    it.each`
+      value                        | expected
+      ${''}                        | ${undefined}
+      ${'api-key:1'}               | ${{ userId: 1 }}
+      ${'service-account:1:admin'} | ${{ userId: 1, login: 'admin' }}
+      ${'user:1:admin'}            | ${{ userId: 1, login: 'admin' }}
+      ${'anonymous:0'}             | ${undefined}
+      ${'render:0'}                | ${undefined}
+      ${':0'}                      | ${undefined}
+    `("parsing '$value' should be '$expected'", ({ value, expected }) => {
+      expect(parseCreatedByValue(value)).toEqual(expected);
```

Comment:
I wonder how stable the format with `:` is. Might be worth considering adding an e2e test that checks if the user is is displayed correctly (might be also a separate task and we add the test before going to private preview)

Reply:
https://github.com/grafana/grafana/issues/89235